project_from_quotation/models/sale_order.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, exceptions, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _compute_project_ids(self):
+        """
+        Override related projects computation to
+        include projects containing tasks related to
+        order lines of this quotation
+        """
+        super()._compute_project_ids()
+        for order in self:
+            quotation_projects = self.env["project.project"].search(
+                [
+                    ("quotation_id", "=", order.id),
+                ]
+            )
+            order.project_ids |= quotation_projects
+            order.project_count = len(order.project_ids)
+        return
+
+    def _compute_tasks_ids(self):
+        """
+        Override related tasks computation to
+        include tasks related to order lines
+        of this quotation
+        """
+        super()._compute_tasks_ids()
+        for order in self:
+            quotation_tasks = self.env["project.task"].search(
+                [
+                    ("quotation_line_id", "in", order.order_line.ids),
+                ]
+            )
+            order.tasks_ids |= quotation_tasks
+            order.tasks_count = len(order.tasks_ids)
+        return
+
+    @api.model
+    def _prepare_task_vals(
+        self,
+        project,
+        order_line,
+    ):
+        """
+        Prepare vals for task creation
+
+        Args:
+        project (project.project): Project
+        order_line (sale.order.line): Order line
+
+        Returns:
+            dict: Dictionary of task values
+        """
+        return {
+            "name": order_line.name,
+            "project_id": project.id,
+            "quotation_line_id": order_line.id,
+            "active": True,
+        }
+
+    @api.model
+    def _prepare_project_vals(self, sale_order):
+        """
+        Prepare vals for project creation
+
+        Args:
+        sale_order (sale.order): Sale Order
+
+        Returns:
+            dict: Dictionary of project values
+        """
+        return {
+            "name": sale_order.name,
+            "quotation_id": sale_order.id,
+            "active": True,
+        }
+
+    def _get_order_lines_unique_products(self, order_lines=None):
+        """
+        Get order lines with unique products
+
+        Args:
+            order_lines (sale.order.line recordset): Order lines
+            if not set, uses self.order_line
+
+        Returns:
+            sale.order.line recordset: Every first
+            line with the same product
+        """
+        self.ensure_one()
+        unique_products = []
+        res_order_lines = self.env["sale.order.line"]
+        order_lines = (order_lines or self.order_line).filtered("product_id")
```

Comment:
What if user doesn't want to create project from all quotation lines and he removes unwanted lines in wizard. This `or` will pull the product that he may have removed already.
We need to use wizard lines only.